var/spack/repos/builtin/packages/libzmq/package.py

```diff
@@ -51,10 +51,10 @@ class Libzmq(AutotoolsPackage):
     depends_on('docbook-xml', type='build', when='+docs')
     depends_on('docbook-xsl', type='build', when='+docs')
 
-    depends_on('libbsd', when='@4.3.3: platform=linux +libbsd')
-    depends_on('libbsd', when='@4.3.3: platform=cray +libbsd')
+    depends_on('libbsd', when='+libbsd')
 
     conflicts('%gcc@8:', when='@:4.2.2')
+    conflicts('+libbsd', when='@:4.3.2')
```

Comment:
Can we use a positive `when` in the variant above and remove the conflict? What I mean is:
```
variant("libbsd", default=(sys.platform != 'darwin'), when='@4.3.3:
```

Reply:
For the record, I'm being pedantic since having conflicts and the variant in might drive clingo insane :laughing: